### PR TITLE
Completed task and removed warnings.

### PR DESCRIPTION
**Pull request by Russyallop**

### CODE REVIEW

File: main.cpp [hunk 1]
```diff
@@ -148,316 +126,135 @@ struct FloatType
 {
     FloatType(float floatIn_)
     {
-        valuePtr = new float;
-        *valuePtr = floatIn_;
-        value = *valuePtr;
+        value = new float(floatIn_);
```
Comment (matkatmusic) [inline main.cpp]: use the member initializer list to init this.
- [x] revise

File: main.cpp [hunk 2]
```diff
@@ -148,316 +126,135 @@ struct FloatType
...
 
 struct DoubleType
 {
     DoubleType(double doubleIn_)
     {
-        valuePtr = new double;
-        *valuePtr = doubleIn_;
-        value = *valuePtr;
-
+        value = new double(doubleIn_);
```
Comment (matkatmusic) [inline main.cpp]: use the member initializer list to init this.
- [x] revise

File: main.cpp [hunk 3]
```diff
@@ -148,316 +126,135 @@ struct FloatType
...
 };
 struct IntType
 {
     IntType(int intIn_)
     {
-
-        valuePtr = new int;
-        *valuePtr = intIn_;
-        value = *valuePtr;
+        value = new int(intIn_);
```
Comment (matkatmusic) [inline main.cpp]: use the member initializer list to init this.
- [x] revise

File: main.cpp [hunk 4]
```diff
@@ -467,181 +264,86 @@ IntType& IntType::divide(int rhs)
...
+    std::cout << "FloatType divide result=" << ( ft.divide( 16.0f) ) << std::endl << std::endl;
 
-    std::cout << "DoubleType add result=" << ( dt.add(2.0).value ) << std::endl;
-    std::cout << "DoubleType subtract result=" << (dt.subtract(2.0).value) << std::endl;
-    std::cout << "DoubleType multiply result=" << (dt.multiply(2.0).value) << std::endl;
-    std::cout << "DoubleType divide result=" << (dt.divide(5.f).value) << std::endl << std::endl;
+    std::cout << "DoubleType add result=" << ( dt.add(2.0) ) << std::endl;
+    std::cout << "DoubleType subtract result=" << (dt.subtract(2.0) ) << std::endl;
+    std::cout << "DoubleType multiply result=" << (dt.multiply(2.0) ) << std::endl;
+    std::cout << "DoubleType divide result=" << (dt.divide(static_cast<double>(5.f) ) )<< std::endl << std::endl;
```
Comment (matkatmusic) [inline main.cpp]: Just change the `float` to a `double` lol!  no need to static_cast that primitive!
- [x] revise

**Review by matkatmusic — CHANGES_REQUESTED** [review]

File: main.cpp (hunk 4 above)
Comment (Russyallop) [reply main.cpp]: Yeah I did think that was weird but it is hard to know what you allow to be changed and what has to stay the same..

**Review by matkatmusic — APPROVED** [review]

**Pull request merged**